src/textParser/tabListener.go

```diff
@@ -0,0 +1,136 @@
+package textParser
+
+import (
+	"fmt"
+	"github.com/antlr4-go/antlr/v4"
+	"strings"
+	"tgScoreBot/src/textParser/parser"
+)
+
+var _ parser.TabListener = (*tabListener)(nil)
+
+type TabProcessor interface {
+	// Bpm sets bpm
+	Bpm(bpm string)
+
+	// Tuning sets guitar tuning
+	Tuning(tuning []string)
+
+	// Duration sets current note duration
+	Duration(duration []string)
+
+	// PlayFret processes tab entry
+	PlayFret(tab string)
+
+	// PlayChord processes a chord
+	PlayChord(chord []string)
+
+	// Pause processes pause
+	Pause()
+
+	// ModelBend models string bending
+	ModelBend(tab string, bendValue string)
+
+	// Errors returns accumulated errors
+	Errors() []error
+
+	// Finish marks the end of parsing
+	Finish()
+}
+
+type tabListener struct {
+	*parser.BaseTabListener
+	processors []*TabProcessor
+	errs       []error
+}
+
+// newTabListener returns a new tabListener given an output and midi clock resolution
+func newTabListener(processors []*TabProcessor) *tabListener {
+	return &tabListener{
+		processors: processors,
+	}
+}
+
+// Implement antlr ErrorListener interface for syntax errors.
+func (s *tabListener) SyntaxError(recognizer antlr.Recognizer, offendingSymbol any, line, column int, msg string, e antlr.RecognitionException) {
+	s.errs = append(s.errs, fmt.Errorf("(%d:%d) %s", line, column, msg))
+}
+
+func (s *tabListener) ExitBpm(ctx *parser.BpmContext) {
+	bpmStr := ctx.GetChild(1).(antlr.ParseTree).GetText()
+	for _, p := range s.processors {
+		(*p).Bpm(bpmStr)
```

Comment:
Здесь и ниже соответственно не должно быть разыменований:
```suggestion
		p.Bpm(bpmStr)
```